extensions/wikia/ImageReview/ImageReviewImageStatus.php

```diff
@@ -78,33 +69,92 @@ function efImageReviewDisplayStatus( ImagePage $imagePage, &$html ) {
 			}
 
 			// oh oh, image is not in queue at all
-			$html .= wfMsg( 'imagereview-imagepage-not-in-queue' );
-
+			$html .= wfMessage( 'imagereview-imagepage-not-in-queue' )->escaped();
 		} else {
 			// image is in the queue but not reviewed yet
-			$html .= wfMsg( 'imagereview-state-0' );
+			$html .= wfMessage( 'imagereview-state-0' )->escaped();
 		}
 	} else {
-		// go through the list and display review states
-
-		while ( $row = $dbr->fetchObject( $res ) ) {
-			$data = array();
-			$data[] = User::newFromId( $row->reviewer_id )->getName();
-			$data[] = wfMsg( 'imagereview-state-' . $row->review_state );
-			$data[] = $row->review_end . ' (UTC)';
-
-			$reviews[] = $data;
-		}
-
 		$html .= Xml::buildTable(
 			$reviews,
-			array(
+			[
 				'class' => 'wikitable filehistory sortable',
 				'style' => 'width: 60%',
-			),
+			],
 			$headers
 		);
 	}
 
 	return true;
 }
+
+function fetchReviewHistory( $dbr, $cityId, $pageId ) {
+	$res = $dbr->select(
+		'image_review_stats',
+		'*',
+		[
+			'wiki_id' => $cityId,
+			'page_id' => $pageId,
+		]
+	);
+
+	$reviews = [];
+	while ( $row = $dbr->fetchObject( $res ) ) {
+		$data = [];
+		$data[] = User::newFromId( $row->reviewer_id )->getName();
+
+		//Possible keys:
+		// imagereview-state-0
+		// imagereview-state-1
+		// imagereview-state-2
+		// imagereview-state-3
+		// imagereview-state-4
+		// imagereview-state-5
+		$data[] = wfMessage( 'imagereview-state-' . $row->review_state )->escaped();
+		$data[] = $row->review_end . ' (UTC)';
+
+		$reviews[] = $data;
+	}
+
+	return $reviews;
+}
+
+function fetchReviewHistoryFromService( $cityId, $pageId, $revisionId ) {
+	$statusMessages = [
+		'UNREVIEWED' => wfMessage( 'imagereview-state-0' )->escaped(),
+		'ACCEPTED' => wfMessage( 'imagereview-state-2' )->escaped(),
+		'QUESTIONABLE' => wfMessage( 'imagereview-state-5' )->escaped(),
+		'REJECTED' => wfMessage( 'imagereview-state-4' )->escaped(),
+		'REMOVED' => wfMessage( 'imagereview-state-3' )->escaped()
+	];
+
+	$reviewHistory = ( new ImageReviewService() )->getImageHistory(
+		$cityId,
+		$pageId,
+		$revisionId,
+		RequestContext::getMain()->getUser()
+	);
+
+	// TODO: filter out initial 'UNREVIEWED' status with user who uploaded revision
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)